feat: add `@stdlib/blas/base/srotg` (#806)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/base/srotg/README.md

<!--

@license Apache-2.0

Copyright (c) 2023 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# srotg

> > Construct a Givens plane rotation for single-precision values `a` and `b`.

Reviewer: ```suggestion
> Construct a Givens plane rotation.
```

> > Constructs a Givens plane rotation for single-precision values `a` and `b`.

Reviewer: ```suggestion
> Construct a Givens plane rotation for single-precision values `a` and `b`.
```

<section class="usage">

## Usage

```javascript
var srotg = require( '@stdlib/blas/base/srotg' );
```

#### srotg( a, b )

> Constructs a Givens plane rotation for single-precision values `a` and `b`.

Reviewer: ```suggestion
Constructs a Givens plane rotation for single-precision floating-point values `a` and `b`.
```

```javascript
var out = srotg( 0.0, 2.0 );
// returns <Float32Array>[ 2.0, 1.0, 0.0, 1.0 ]
```

The function has the following parameters:

-   **a**: rotational elimination parameter.
-   **b**: rotational elimination parameter.

#### srotg.assign( a, b, out, stride, offset )

> Constructs a Givens plane rotation for single-precision floating-point values and assigns results to an output array.

Reviewer: ```suggestion
Constructs a Givens plane rotation for single-precision floating-point values `a` and `b` and assigns results to an output array.
```

> Constructs a Givens plane rotation for single-precision floating-point values and assigns results to an output array.

Reviewer: ```suggestion
Constructs a Givens plane rotation for single-precision floating-point values `a` and `b` and assigns results to an output array.
```

<!--

@license Apache-2.0

Copyright (c) 2023 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# srotg

> > Constructs a Given plane rotation for single-precision values `a` and `b`.

Reviewer: ```suggestion
> Constructs a Givens plane rotation for single-precision values `a` and `b`.
```

Should be updated here and below.

<section class="usage">

## Usage

```javascript
var srotg = require( '@stdlib/blas/base/srotg' );
```

#### srotg( a, b )

Constructs a Given plane rotation for single-precision values `a` and `b`.

```javascript
var Float32Array = require( '@stdlib/array/float32' );

var a = 0.0;
var b = 2.0;

var out = srotg( a, b );
// returns <Float32Array>[ 2.0, 1.0, 0.0, 1.0 ]
```

The function has the following parameters:

-   **a**: rotational elimination parameter.
-   **b**: rotational elimination parameter.

> </section>

Reviewer: We should also document the `assign` API as it is part of the public interface.

<!--

@license Apache-2.0

Copyright (c) 2023 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# srotg

> Constructs a Givens plane rotation for single-precision values `a` and `b`.

<section class="usage">

## Usage

```javascript
var srotg = require( '@stdlib/blas/base/srotg' );
```

#### srotg( a, b )

Constructs a Givens plane rotation for single-precision values `a` and `b`.

```javascript

> var Float32Array = require( '@stdlib/array/float32' );
> 

Reviewer: ```suggestion
```

This import is not used.

> var a = 0.0;
> var b = 2.0;
> 
> var out = srotg( a, b );

Reviewer: ```suggestion
var out = srotg( 0.0, 2.0 );
```

// returns <Float32Array>[ 2.0, 1.0, 0.0, 1.0 ]
```

The function has the following parameters:

-   **a**: rotational elimination parameter.

> -   **b**: rotational elimination parameter.

Reviewer: These both have the same description?

Author: Yeah.

#### srotg.assign( a, b, out, stride, offset )

> Constructs Givens plane rotations for single-precision floating-point values and assigns results to output array.

Reviewer: ```suggestion
Constructs a Givens plane rotation for single-precision floating-point values and assigns results to an output array.
```

```javascript
var Float32Array = require( '@stdlib/array/float32' );

var out = new Float32Array( 4 );

var y = srotg.assign( 0.0, 2.0, out, 1, 0 );
// returns <Float32Array>[ 2.0, 1.0, 0.0, 1.0 ]

var bool = ( y === out );
// returns true
```

</section>

<!-- /.usage -->

<section class="notes">

## Notes

-   `srotg()` corresponds to the [BLAS][blas] level 1 function [`srotg`][srotg].

</section>

<!-- /.notes -->

<section class="examples">

```javascript
var randu = require( '@stdlib/random/base/randu' );
var round = require( '@stdlib/math/base/special/round' );
var srotg = require( '@stdlib/blas/base/srotg' );

var out;
var a;
var b;
var i;

for ( i = 0; i < 100; i++ ) {

>     a = round( randu()*10.0 ) - 5.0;

Reviewer: Instead of `round(randu())`, let's use `discreteUniform( -5, 5 )`.

    b = round( randu()*10.0 ) - 5.0;
    out = srotg( a, b );
    console.log( out );
}
```

</section>

<!-- /.examples -->

<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->

> <section class="related">
> 
> * * *
> 
> ## See Also
> 
> -   <span class="package-name">[`@stdlib/blas/base/daxpy`][@stdlib/blas/base/daxpy]</span><span class="delimiter">: </span><span class="description">multiply a vector x by a constant and add the result to y.</span>
> -   <span class="package-name">[`@stdlib/blas/base/dscal`][@stdlib/blas/base/dscal]</span><span class="delimiter">: </span><span class="description">multiply a double-precision floating-point vector by a constant.</span>
> -   <span class="package-name">[`@stdlib/blas/base/gscal`][@stdlib/blas/base/gscal]</span><span class="delimiter">: </span><span class="description">multiply a vector by a constant.</span>
> -   <span class="package-name">[`@stdlib/blas/base/saxpy`][@stdlib/blas/base/saxpy]</span><span class="delimiter">: </span><span class="description">multiply a vector x by a constant and add the result to y.</span>
> 

Reviewer: ```suggestion
<section class="related">

```

</section>

<!-- /.related -->

<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="links">

[blas]: http://www.netlib.org/blas

[srotg]: http://www.netlib.org/lapack/explore-html/df/d28/group__single__blas__level1.html

> 
> <!-- <related-links> -->
> 
> [@stdlib/blas/base/daxpy]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/base/daxpy
> 
> [@stdlib/blas/base/dscal]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/base/dscal
> 
> [@stdlib/blas/base/gscal]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/base/gscal
> 
> [@stdlib/blas/base/saxpy]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/base/saxpy
> 
> <!-- </related-links> -->

Reviewer: ```suggestion
```
